src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -28,15 +28,37 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.server")
+	log               = golog.LoggerFor("flashlight.server")
+	frontingProviders = map[string]func(*http.Request) bool{
+		// WARNING: If you add a provider here, keep in mind that Go's http
+		// library normalizes all header names so the first letter of every
+		// dash-separated "word" is uppercase while all others are lowercase.
+		// Also, try and check more than one header to lean on the safe side.
+		"cloudflare": func(req *http.Request) bool {
+			return hasHeader(req, "Cf-Connecting-Ip") || hasHeader(req, "Cf-Ipcountry") || hasHeader(req, "Cf-Ray") || hasHeader(req, "Cf-Visitor")
+		},
+		"cloudfront": func(req *http.Request) bool {
+			return hasHeader(req, "X-Amz-Cf-Id") || headerMatches(req, "User-Agent", "Amazon Cloudfront")
+		},
+	}
```

Comment:
So maybe the best option would be that clients send this custom header to declare front, and the server uses if it present, falling back to checking CDN-specific headers if not.  I'm inclined to go YAGNI on this, since it seems so unlikely to ever make a difference, and there's some desirable simplicity in leaving the client out of this.